sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricCell.java

```diff
@@ -28,7 +29,17 @@
  * @param <DataT> The type of metric data stored (and extracted) from this cell.
  */
 @Experimental(Kind.METRICS)
-public interface MetricCell<UserT extends Metric, DataT> {
+public interface MetricCell<UserT extends Metric, DataT> extends Serializable {
+
+  /**
+   * Update value of this cell.
+   */
+  void update(DataT data);
```

Comment:
The thing is it isn't a `set` it combines the input with the existing value, which is why I opted for `update` as the name.
We could change this to `combineAndSet` or `mergeAndSet` if that is clearer.